keras-dreambooth-sprint/README.md

```diff
@@ -0,0 +1,155 @@
+prompt = f"A photo of {unique_id} {class_label} in a bucket"
+image = pipeline(prompt, num_inference_steps=50).images[0]
+```
+
+### **Model Hosting**
+
+At the end of this notebook you will see a section dedicated for hosting, and a separate one for inference. We will be using huggingface_hub library’s Keras-specific model pushing and loading functions, `push_to_hub_keras` and `from_pretrained_keras` . We will first push the model using `push_to_hub_keras`. After model is pushed, you will see the model is hosted with a model card like below 👇 
+
+![Repository](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/dreamboothrepo.png)
+
+to version the models better, enable discoverability and reproducibility, we will fill the model card.  Click `Edit model card` . We will first fill the Metadata section of the model card. If your model is trained with a dataset from Hugging Face Hub, you can fill the datasets section with the dataset. We will provide fill `pipeline_tag` with `Text-to-Image` and pick a license for our model. 
+
+![Metadata](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/dreambooth-etadata.png)
+
+Then, we will fill the markdown part. Hyperparameters and plot is automatically generated so we can write a short explanation for description, intended use and dataset.
+
+
+You can find the example repository below 👇 
+
+[keras-dreambooth/dreambooth_diffusion_model · Hugging Face](https://huggingface.co/keras-dreambooth/dreambooth_diffusion_model)
+
+### Model Demo
+
+We will use Gradio to build our demos for the models we have trained. With `Interface` class it’s straightforward 👇  
+
+```python
+import gradio as gr
+
+# write function for inference
+def infer(prompt):
+		# assume you loaded the model
+    generated_images = sd_dreambooth_model.text_to_image(
+        prompt, batch_size=4
+    )
+    return generated_images 
+    
+output = gr.Gallery(label="Outputs").style(grid=(2,2))
+
+# pass function, input type for prompt, the output for multiple images
+gr.Interface(infer, inputs=["text"], outputs=[output]).launch()
+```
+
+You can check out `app.py`file of the application below and repurpose it for your model!
+
+[Dreambooth Submission - a Hugging Face Space by keras-dreambooth](https://huggingface.co/spaces/keras-dreambooth/example-submission)
+
+This app generates images of a corgi 🐶 
+
+![Dreambooth App](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/dreambooth_corgi.png)
+
+**Creating a Space**
+
+After our application is written, we can create a Hugging Face Space to host our app. You can go to [huggingface.co](http://huggingface.co), click on your profile on top right and select “New Space”.
+
+![New Space](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/new_space.png)
+
+
+We can name our Space, pick a license and select Space SDK as “Gradio”. Free hardware is enough for our app, so no need to change it.
```

Comment:
I thought this required GPU no?

Reply:
Yes. 

If you're using a T4, you should enable mixed-precision by default. A complete demo with a fine-tuned Stable Diffusion model from KerasCV is available here: https://huggingface.co/spaces/sayakpaul/pokemon-sd-kerascv

Reply:
Note that this will require people to upgrade a Space. Maybe we can use a single space to load them all, as done for the official dreambooth library space